### PR TITLE
Fix: ocean api tx get vouts

### PR DESCRIPTION
**Pull request by canonbrother**
## Summary

- as per title

#### checked
```
getVouts
    ✓ should return list of vout (11 ms)
    ✓ should return empty page if txid is not valid (2 ms)
    ○ skipped should return list of vout when next is out of range
```

### CODE REVIEW

File: lib/ain-ocean/src/api/transactions.rs [hunk 1]
```diff
@@ -74,11 +74,18 @@ async fn get_vouts(
     Query(query): Query<PaginationQuery>,
     Extension(ctx): Extension<Arc<AppContext>>,
 ) -> Result<ApiPagedResponse<TransactionVout>> {
+    let next = query
+        .next
+        .clone()
+        .unwrap_or(format!("0"))
+        .parse::<usize>()?;
```
Comment (Jouzo) [inline lib/ain-ocean/src/api/transactions.rs:81]:
```suggestion
let next = query
    .next
    .as_ref()
    .map(String::as_str)
    .unwrap_or("0")
    .parse::<usize>()?;
```

File: lib/ain-ocean/src/model/transaction_vout.rs [hunk 2]
```diff
@@ -1,10 +1,9 @@
 use bitcoin::Txid;
 use serde::{Deserialize, Serialize};
 
-pub type TransactionVoutKey = (Txid, usize);
-
 #[derive(Debug, Serialize, Deserialize)]
 pub struct TransactionVout {
+    pub id: String, // format!({}{:08x}, txid, usize")
```
Comment (Jouzo) [inline lib/ain-ocean/src/model/transaction_vout.rs:6]: Why are we reverting back to String here ?

Comment (canonbrother) [reply lib/ain-ocean/src/model/transaction_vout.rs:6]: the old one usize wasn't padded
references: https://github.com/BirthdayResearch/jellyfishsdk/blob/767aa7199458153486350a6804fa0851932d91ae/apps/whale-api/src/module.model/_hex.encoder.ts#L37

File: lib/ain-ocean/src/api/transactions.rs (hunk 1 above)
Comment (canonbrother) [reply lib/ain-ocean/src/api/transactions.rs]: fixed at [f06e14e](https://github.com/DeFiCh/ain/pull/2834/commits/f06e14ea39dd7af7f86a961499224c38743375e7)

clippy updates:
```
let next = query
        .next
        .as_deref()
        .unwrap_or("0")
        .parse::<usize>()?;
```

File: lib/ain-ocean/src/model/transaction_vout.rs (hunk 2 above)
Comment (Jouzo) [reply lib/ain-ocean/src/model/transaction_vout.rs:6]: We should fully implement `Column` trait for `TransactionVout` column instead

Comment (Jouzo) [reply lib/ain-ocean/src/model/transaction_vout.rs:6]: Can check `TransactionByBlockHash` implementation for ref

File: lib/ain-ocean/src/api/transactions.rs (hunk 1 above)
Comment (Jouzo) [reply lib/ain-ocean/src/api/transactions.rs]: Even better 👍

File: lib/ain-ocean/src/model/transaction_vout.rs (hunk 2 above)
Comment (canonbrother) [reply lib/ain-ocean/src/model/transaction_vout.rs]: fixed at [feac743](https://github.com/DeFiCh/ain/pull/2834/commits/feac74377bf6dc59556672799e6c9d744ec132ed)

**Pull request merged**